movie-browser/src/App.js

```diff
@@ -1,31 +1,61 @@
 /* eslint-disable no-unused-vars */
 import React from 'react'
-import CardContainer from './CardContainer'
+import CardContainer from './view/CardContainer'
+import FeaturedMovie from './view/FeaturedMovie'
 import ApiReader from './model/ApiReader';
 
 // *** constant(s)
 const url = 'https://ghibliapi.herokuapp.com/films'
+const btnTxtDef = "View all"
+
+// TODO: fix sorting by year. have no idea why it works for score but not for year
+// TODO: makes button working
 
 // *** Actual App
 class App extends React.Component {
-  state = {movies: []};
+  state = {movies: [], featuredMovieIndex: 0, btnTxt: btnTxtDef};
 
   // *** running ApiReader when mounted
   componentDidMount() {
     this.fetchData();
   }
 
+  pickRandomMovieIndex = () => this.setState({
+    featuredMovieIndex: Math.floor(Math.random() * this.state.movies.length)
```

Comment:
you can retrieve this on apireader as well. You don't need to return only one thing from your apireader. You can return an object with several things.

Reply:
Same as above...
I haven't thought I can use the object literal this way.
Good to know.